docs/proposals/202106120-edgex-integration.md

```diff
@@ -0,0 +1,144 @@
+
+## Proposal
+
+### User Stories
+
+1. As an end-user, I would like to deploy EdgeX to the Edge node
+2. As an end-user, I would like to deploy my container based application to the Edge node and connect to the EdgeX I deployed.
+3. As an end-user, I would like to update the EdgeX (add/remove one or more optional components).
+
+
+### Overview
+
+We define a new CR EdgeX to represent the EdgeX instance running on the Edge side, the relation between the EdgeX CR and EdgeX instance is 1:1. When we have multiple EdgeX instances running on the Edge side, the same number of EdgeX CRs will be created. For each EdgeX instance it may contains different micro-service components, each of the components will be live in the form of a Kubernetes deployment. For the deployments that compose a EdgeX instance it can basically grouped into two categories: 
+- Required deployments : For the deployments in this group are mandatory and will be deployed with each EdgeX instance.
+- Optional deployments: For deployments in this group is optional can user can specify which deployments they want to deploy.
+
+Consider it is quite common to have multiple EdgeX instance in a single Kubernetes Cluster, With traditional service+deployment solutions, for the components in each EdgeX instance it require a unique name. When the EdgeX instance number increased the name management will be a critical issue need to address. In this proposal we leverage the United Deployment feature of OpenYurt. In that way all the EdgeX Instance share the same service name which can avoid the service name explosion.
+
+![Deployment Topology](../img/edgex/overview.png)
+
+As the deployment topology diagram shows.  Each components in the EdgeX instance will have a corresponding service mapping to it. The service will mapping to a United Deployment instance(In the backend it is actually mapping to all the deployments inside the united deployment) On the edge side a deployment with a unique name will be created and registered in the service's endpoints. Service topology policy will be automatically applied by United Deployment controller to make sure the traffic will be limited to Edge/Node Pool level.From one side we can make the management easier and on the other side we can still keep the EdgeX instance isolation. 
+#### 1. EdgeX Customer Resource Definition
+
+Following is the definition of `EdgeX` CRD and its related Golang struct. The CRD will be created when the EdgeX controller is up and running. 
+```yaml
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    ...
+  creationTimestamp: null
+  name: edgex.apps.openyurt.io
+spec:
+  ...
+  optionalComponents:
+    description: 
+    type: object
+
+```
+
+
+```go
+type EdgeX struct {
+    //TODO
```

Comment:
I think we need add selector of nodePool into `EdgeX struct`

Reply:
yes, Wuming is working on the EdgeX struct definition stuff.

Reply:
ok